print error messages on boot failure due to bad pill

joemfb opened the pull request:
Advice welcome wrt clearer, more specific error messages.

closes #1162

belisarius222 commented on noun/manage.c:
```diff
@@ -1708,13 +1717,39 @@ u3m_boot(c3_o nuu_o, c3_o bug_o, c3_c* dir_c,
+      {
+        u3_noun pro = u3m_soft(0, u3ke_cue, u3k(pil));
+
+        if ( 0 != u3h(pro) ) {
```
Is it ok to use a raw zero here, instead of `u3_blit` or whatever?

joemfb replied:
I don't think there's a more specific convention for checking the tag of a bail noun (a `+tone` or  `+toon`). The success case is `[%0 p=*]`, so there won't be any practical issue.

belisarius222 replied:
Ah, it's a noun zero. Ok, that's fine.

joemfb commented:
This PR is a great example of why profiling is turned on in CI ... And now it looks like something's leaking. I'm looking for it.

joemfb commented:
I fixed the first leak (that I had introduced in `u3m_soft`), and avoided the second by virtualizing more granularly.

@belisarius222, please re-review.

joemfb commented on noun/manage.c:
```diff
@@ -1180,28 +1180,40 @@ u3m_soft(c3_w    sec_w,
+    //
+    if ( 0 == u3A->roc ) {
+      u3z(why);
+      return u3nc(2, u3_nul);
```
On second thought, this doesn't seem to be entirely consistent with the contract of `u3m_soft`. In the normal case, errors are produced with a specific mote instead of the original numeric tag. The success case is still consistent, and these changes are only being made to support soft-ing without an installed kernel, which is a rare case. I'm comfortable with the inconsistency, but i'll leave a comment explaining it further.
